src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,241 @@
+        this.setState({shouldShowArrow});
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const sourceURL = _.get(attachmentItem, 'sourceURL', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        return {
+            sourceURL: addEncryptedAuthTokenToURL(sourceURL),
+            file,
+        };
+    }
+
+    /**
+     * This is called when there are new reports to set the state
+     */
+    makeStateWithReports() {
+        let page;
+        const actionsArr = ReportActionsUtils.getSortedReportActions(this.props.reportActions);
+        const attachments = _.reduce(actionsArr, (attachmentsAccumulator, {action: reportAction}) => {
+            if (reportAction.originalMessage && reportAction.originalMessage.html) {
+                const matchesIt = reportAction.originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA);
+                const matches = [...matchesIt];
+                if (matches.length === 2) {
+                    const [src, name] = matches;
+                    if ((this.state.sourceURL && src[2].includes(this.state.sourceURL))
+                        || (!this.state.sourceURL && src[2].includes(this.props.sourceURL))) {
+                        page = attachmentsAccumulator.length;
+                    }
+                    const {pathname} = new URL(src[2]);
+                    attachmentsAccumulator.push({sourceURL: pathname, file: {name: name[2]}});
+                }
+            }
+            return attachmentsAccumulator;
+        }, []);
+
+        const {sourceURL, file} = this.getAttachment(attachments[page]);
+        this.setState({
+            page,
+            attachments,
+            sourceURL,
+            file,
+            isForwardDisabled: page === 0,
+            isBackDisabled: page === attachments.length - 1,
+        });
+    }
+
+    /**
+     * Retrieves the next set of report actions for the chat once we are nearing the end of what we are currently
+     * displaying.
+     */
+    loadMoreChats() {
+        // Only fetch more if we are not already fetching so that we don't initiate duplicate requests.
+        if (this.props.report.isLoadingMoreReportActions) {
+            return;
+        }
+
+        const minSequenceNumber = _.chain(this.props.reportActions)
+            .pluck('sequenceNumber')
+            .min()
+            .value();
+
+        if (minSequenceNumber === 0) {
+            return;
+        }
+
+        // Retrieve the next REPORT.ACTIONS.LIMIT sized page of comments, unless we're near the beginning, in which
+        // case just get everything starting from 0.
+        const oldestActionSequenceNumber = Math.max(minSequenceNumber - CONST.REPORT.ACTIONS.LIMIT, 0);
+        Report.readOldestAction(this.props.report.reportID, oldestActionSequenceNumber);
+    }
```

Comment:
This is not DRY, we should have invoked which is already defined. But that might require a refactor what do you think @chiragsalian

Reply:
Correct yes we should invoke what is already defined to DRY the code.
@JediWattson, can you update such that,
- move loadMoreChats to actions/Report.js. (maybe rename it to loadMoreActions to make it clearer)
- You can accept the param isLoadingMoreReportActions and reportActions to do the same functionality
- Have AttachmentCarousel and ReportActionsView just call the method Report.loadMoreChats.

This would DRY up the code.